src/Tasks/AssemblyDependency/HintPathResolver.cs

```diff
@@ -51,7 +51,7 @@ public override bool Resolve
             out bool userRequestedSpecificFile
         )
         {
-            if (!string.IsNullOrEmpty(hintPath))
+            if (!string.IsNullOrEmpty(hintPath) && !FileUtilities.PathIsInvalid(hintPath))
```

Comment:
I think this deserves a comment. Probably with a link to the bug to consider trimming HintPath in 17.0.

Reply:
Added comment, https://github.com/microsoft/msbuild/pull/4598/commits/fc762c4b478fe94d329bf15b907da88e8e2dfb92

and I do need to have ask mode write up